cmd/globals.go

```diff
@@ -200,5 +204,21 @@ func setGlobalsFromContext(ctx *cli.Context) error {
 			globalResolvers[host] = addr
 		}
 	}
+
+	customHeaders := ctx.StringSlice("custom-header")
+	if len(customHeaders) > 0 {
+		globalCustomHeader = make(http.Header)
+		for _, header := range customHeaders {
+			i := strings.IndexByte(header, ':')
+			if i <= 0 {
+				return fmt.Errorf("invalid custom header entry %s", header)
+			}
+			if !httpguts.ValidHeaderFieldName(header[:i]) || !httpguts.ValidHeaderFieldValue(header[i+1:]) {
+				return fmt.Errorf("invalid custom header entry %s", header)
+			}
+			globalCustomHeader.Add(header[:i], header[i+1:])
```

Comment:
Your change does check the header value, but still has the trailing space.
```suggestion
			h := strings.TrimSpace(header[:i])
			hv := strings.TrimSpace(header[i+1:])
			if !httpguts.ValidHeaderFieldName(h) || !httpguts.ValidHeaderFieldValue(hv) {
				return fmt.Errorf("invalid custom header entry %s", header)
			}
			globalCustomHeader.Add(h, hv)
```